sql/sql_delete.cc

```diff
@@ -1423,29 +1490,57 @@ int multi_delete::do_deletes()
    @retval  1 Triggers or handler reported error.
    @retval -1 End of file from handler.
 */
-int multi_delete::do_table_deletes(TABLE *table, SORT_INFO *sort_info,
-                                   bool ignore)
+int multi_delete::rowid_table_deletes(TABLE *table, bool ignore)
 {
   int local_error= 0;
-  READ_RECORD info;
   ha_rows last_deleted= deleted;
-  DBUG_ENTER("do_deletes_for_table");
-
-  if (unlikely(init_read_record(&info, thd, table, NULL, sort_info, 0, 1,
-                                FALSE)))
-    DBUG_RETURN(1);
+  DBUG_ENTER("rowid_table_deletes");
+  TABLE *err_table= nullptr;
 
   bool will_batch= !table->file->start_bulk_delete();
-  while (likely(!(local_error= info.read_record())) && likely(!thd->killed))
+  TABLE *tmp_table= tmp_tables[table_being_deleted->shared];
+  tmp_table->file->extra(HA_EXTRA_CACHE);  // Change to read cache
+  if (unlikely((local_error= table->file->ha_rnd_init(0))))
+  {
+    err_table= table;
+    goto err;
+  }
+  table->file->extra(HA_EXTRA_NO_CACHE);
+  if (unlikely((local_error= tmp_table->file->ha_rnd_init(1))))
   {
+    err_table= tmp_table;
+    goto err;
+  }
+
+  while (!thd->killed) {
```

Comment:
Coding style here and in code around:  `{` should be on its own line...

Reply:
Fixed.